src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs

```diff
@@ -69,10 +69,35 @@ public TimelineViewModel()
                 SelectedForEditTEId = open ? te.GUID : SelectedForEditTEId;
             this.WhenAnyValue(x => x.SelectedForEditTEId, x => x.TimeEntryBlocks)
                 .ObserveOn(RxApp.TaskpoolScheduler).Subscribe(_ =>
-                TimeEntryBlocks?.ForEach(te => te.IsEditViewOpened = SelectedForEditTEId == te.TimeEntryId));
+                    TimeEntryBlocks?.ForEach(te => te.IsEditViewOpened = SelectedForEditTEId == te.TimeEntryId));
+            this.WhenAnyValue(x => x.SelectedForEditTEId, x => x.RunningTimeEntryBlock)
+                .ObserveOn(RxApp.TaskpoolScheduler).Where(pair => pair.Item2 != null)
+                .Subscribe(pair => pair.Item2.IsEditViewOpened = pair.Item1 == pair.Item2.TimeEntryId);
             Observable.Timer(TimeSpan.Zero,TimeSpan.FromMinutes(1))
                 .Select(_ => ConvertTimeIntervalToHeight(DateTime.Today, DateTime.Now, SelectedScaleMode))
                 .Subscribe(h => CurrentTimeOffset = h);
+            var timeEntryBlocksObservable = this.WhenAnyValue(x => x.TimeEntryBlocks);
+            Observable.FromEvent<Toggl.DisplayRunningTimerState, Toggl.TogglTimeEntryView>(handler =>
+                    {
+                        void h(Toggl.TogglTimeEntryView te) => handler(te);
+                        return h;
+                    },
+                    handler => Toggl.OnRunningTimerState += handler,
+                    handler => Toggl.OnRunningTimerState -= handler)
+                .CombineLatest(timeEntryBlocksObservable, (te, tes) => ConvertToRunningTimeEntryBlock(te, RunningTimeEntryBlock, tes, SelectedScaleMode, CurrentTimeOffset))
+                .Subscribe(te => RunningTimeEntryBlock = te);
+            Observable.FromEvent<Toggl.DisplayStoppedTimerState, Unit>(handler =>
+                    {
+                        void h() => handler(Unit.Default);
+                        return h;
+                    },
+                    handler => Toggl.OnStoppedTimerState += handler,
+                    handler => Toggl.OnStoppedTimerState -= handler)
+                .Subscribe(_ => RunningTimeEntryBlock = null);
+            this.WhenAnyValue(x => x.CurrentTimeOffset).Where(_ => RunningTimeEntryBlock != null).Subscribe(curOffset =>
+                RunningTimeEntryBlock.Height = Math.Max(curOffset - RunningTimeEntryBlock.VerticalOffset - 1, TimelineConstants.MinTimeEntryBlockHeight));
+            this.WhenAnyValue(x => x.TimeEntryBlocks, x => x.RunningTimeEntryBlock, x => x.IsTodaySelected)
+                .Select(item => item.Item1?.Any() == true || (item.Item2 != null && item.Item3)).ToPropertyEx(this, x => x.AnyTimeEntries);

```

Comment:
You can use a selector function as the fourth parameter to WhenAnyValue. This way you can avoid ugly variable naming (item.Item1|2|3) and make the code much more readable. This might also be slightly better in terms of efficiency/performance.